retail-managment-system/rms-ui/src/app/core/services/resource.service.ts

```diff
@@ -0,0 +1,96 @@
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+import { SettingsService } from './settings.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export abstract class ResourceService<T> {
+  protected readonly APIUrl =
+    SettingsService.configurationEnvironment.api.baseUrl +
+    this.getResourceUrl();
+
+  protected constructor(protected httpClient: HttpClient) {}
+
+  abstract getResourceUrl(): string;
+
+  toServerModel(entity: T): any {
+    return entity;
+  }
+
+  fromServerModel(json: any): T {
+    return json;
+  }
+
+  getList(p: {} = {}): Observable<T> {
```

Comment:
return in array here you receive it as just one.
it will not work if you used endpoint and access data returned